src/storage-controller/src/collection_mgmt.rs

```diff
@@ -1153,6 +1049,55 @@ where
     (tx, handle.abort_on_drop(), shutdown_tx)
 }
 
+async fn monotonic_append<T: Timestamp + Lattice + Codec64 + TimestampManipulation>(
```

Comment:
I'm not entirely sure I ported the intention here correctly, it was kinda hard to follow